sklearn/_config.py

```diff
@@ -134,6 +136,12 @@ def set_config(
 
         .. versionadded:: 1.2
 
+    skip_parameter_validation : bool, default=None
+        If True, parameter validation of public estimators and functions will be
```

Comment:
For me it wasn't immediately clear what "parameter validation of public estimators and functions" covers and what it doesn't covers. After a look at the code I think it only covers the constructor arguments of estimators and the arguments passed to functions. Where as before I looked at the code I thought it might also disable validation of the input data (I think it doesn't really do that).

So I was thinking about alternative sentences. Here a few ideas. Maybe you like some of them or we can combine them.

- If True, disable the validation of estimator's hyper-parameters and arguments to helper functions. This can be useful for very small inputs, but it can lead to crashes.
- If True, disable the validation of the hyper-parameter values of estimators and arguments passed to other functions. This can be useful in situations like small input data, but it can also lead to misconfigurations going undetected.

Reply:
I adapted the description to be more explicit about what kind of validation is disabled. I did not keep the `like small input data` part because it can be problematic even with large data if you have validation in a loop that processes samples one at a time for instance.

Reply:
Sounds good. In my head I had implicitly set "dataset size == number of samples passed to predict", so the case you described is part of that. However, we have at least N=1 data that this "obvious" assumption isn't obvious to everyone :D So I'm ok with this. (I can't mark this as resolved, but if I could I would)

Reply:
> "dataset size == number of samples passed to predict",

What I meant is that here predict might call a public function for each sample which would be catastrophic